src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTestBase.cs

```diff
@@ -124,4 +128,49 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
             }
         }
     }
+
+    public static class HttpClientExtensions
+    {
+        public static Task<HttpResponseMessage> SendAsync(this HttpClient client, bool async, HttpRequestMessage request, HttpCompletionOption completionOption = default, CancellationToken cancellationToken = default)
+        {
+            if (async)
+            {
+                return client.SendAsync(request, completionOption, cancellationToken);
+            }
+            else
+            {
+#if NETCOREAPP
```

Comment:
We've generally put such functionality into platform-specific files, e.g. *.netcoreapp.cs and *.netstandard.cs, rather than using ifdefs.